eodatasets3/prepare/landsat_l1_prepare.py

```diff
@@ -235,23 +284,26 @@ def prepare_and_write(
 
     Input dataset path can be a folder or a tar file.
     """
-    mtl_doc, mtl_filename = get_mtl_content(ds_path)
+    mtl_doc, root_element, mtl_filename = get_mtl_content(ds_path)
     if not mtl_doc:
         raise ValueError(f"No MTL file found for {ds_path}")
-
-    usgs_collection_number = mtl_doc["metadata_file_info"].get("collection_number")
+    coll = "C2" if root_element == "landsat_metadata_file" else "C1"
+    coll_map = LANDSAT_MTL_MAP[coll]
+    usgs_collection_number = mtl_doc[coll_map["product_contents_cn"]].get(
+        "collection_number"
+    )
```

Comment:
I'm slightly iffy about this first `if` statement, as more collections will be released in the future (... and already, pre-collection) that will quietly identify as the wrong collection number from it.

The `usgs_collection_number` below it contains the same information (but as an int), as is calculated from actual properties, so maybe we could just reuse that instead of this extra `coll` variable? (it also has a clearer name :slightly_smiling_face:)

Reply:
Yes, if there are more collections the if statement will need to change.   Using usgs_collection_number is tough since it relies on the collection_key to get its value.  I've changed coll to collection_key, to use a clearer name.